probe-rs/src/probe/cmsisdap/tools.rs

```diff
@@ -266,33 +316,61 @@ pub fn open_device_from_selector(
     );
 
     // Attempt to open provided VID/PID/SN with hidapi
+
+    let hid_api = HidApi::new()?;
+
+    let mut device_list = hid_api.device_list();
+
+    let device_info = device_list
+        .find(|info| {
+            let mut device_match = info.vendor_id() == vid && info.product_id() == pid;
+
+            if let Some(sn) = sn {
+                device_match &= Some(sn.as_ref()) == info.serial_number();
+            }
+
+            if let Some(hid_interface) = selector.usb_hid_interface {
+                device_match &= info.interface_number() == hid_interface as i32;
+            }
+
+            device_match
+        })
+        .ok_or(ProbeCreationError::NotFound)?;
+
+    let device = device_info.open_device(&hid_api)?;
+
+    /*
```

Comment:
Any reason to keep this old code here commented out?

Reply:
Just forgot to remove it, thanks for catching this.